### PR TITLE
tests: kernel: early_sleep: move to new ztest API

### PR DESCRIPTION
**Pull request by ghost**
Move tests/kernel/early_sleep/ to use new ztest API.

Signed-off-by: Guo Lixin <lixinx.guo@intel.com>

### CODE REVIEW

**Comment by LixinGuoX** [comment]
We have the same issue in #47126 on qemu_cortex_m3.

**Review by andyross — DISMISSED** [review]
FWIW: this is testing a "feature" (the ability to suspend execution during single threaded kernel initialization before main() is entered!) that AFAIK we don't actually document and that I'd prefer we not even have.  I occasionally notice it and have a "Why?!!" moment.  Would be nice to see it die and have a firm "don't block initialization" rule...

Obviously there's nothing wrong with the ztest port though.

**Review by nashif — APPROVED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
usage of 1cpu is wrong here.

File: tests/kernel/early_sleep/src/main.c [hunk 1]
```diff
@@ -91,7 +91,7 @@ SYS_INIT(test_early_sleep_app, APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
  *
  * @see k_sleep()
  */
-static void test_early_sleep(void)
+ZTEST(earlysleep_1cpu, test_early_sleep)
```
Comment (nashif) [inline tests/kernel/early_sleep/src/main.c:94]: no need for 1cpu in the suite name.

Comment (LixinGuoX) [reply tests/kernel/early_sleep/src/main.c]: You mean rename the suite name if it only contains cases of 1cpu?
We also have combo scenario that both have `ztest_1cpu_unit_test` and `ztest_unit_test`, like kernel/context. See #47229 ,is it ok to change like that PR?

**Review by dcpleung — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Comment by LixinGuoX** [comment]
Hi, @nashif , could you help to review again?

**Comment by enjiamai** [comment]
Hi @nashif , could you please re-review this when you get a chance? Thank you!

**Review by dcpleung — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**